subsys/net/lib/app/client.c

```diff
@@ -379,12 +379,43 @@ int net_app_init_client(struct net_app_ctx *ctx,
 	}
 
 	if (client_addr) {
-		memcpy(&addr, client_addr, sizeof(addr));
+		u16_t local_port = 0;
+		bool empty_addr = false;
 
-		if (addr.sa_family != remote_addr.sa_family) {
-			NET_DBG("Address family mismatch %d vs %d",
-				addr.sa_family, remote_addr.sa_family);
-			return -EINVAL;
+		addr.sa_family = remote_addr.sa_family;
+
+		/* local_port is used if the IP address isn't specified */
+#if defined(CONFIG_NET_IPV4)
```

Comment:
@jukkar This is the code I was referring to where the remote_addr family doesn't matter.  It's a bit sloppy for sure, but without changing the API from a sockaddr (client_addr) to a u16_t port, this is the best I could come up with.